src/osd/ReplicatedPG.cc

```diff
@@ -8864,6 +8864,7 @@ void ReplicatedPG::on_change(ObjectStore::Transaction *t)
 
   // requeues waiting_for_active
   scrub_clear_state();
+  scrub_unreserve_replicas();
```

Comment:
The replica must clear its reservation on its own when it goes through on_change.  Sending the unreservation messages here is actually wrong since it would be tagged with an epoch from the *next* interval.  We might not even be primary in that interval.